### ISSUE TITLE
Meson unable to find shared object known by ldconfig.

### ISSUE THREAD

**Issue by c-edw** [mesonbuild/meson#4375]
I have added `/usr/local/lib64` to the `ldconfig` configuration files and ran `ldconfig` to update the cache. I have confirmed that the shared objects have been picked up by `ldconfig` by running:
```
$ ldconfig -p | grep wlroots
        libwlroots.so.0 (libc6,x86-64) => /usr/local/lib64/libwlroots.so.0
        libwlroots.so (libc6,x86-64) => /usr/local/lib64/libwlroots.so
```

However, running `meson build` in a project which requires this library outputs the following:
```
meson.build:33:0: ERROR:  Dependency "wlroots" not found, tried pkgconfig
```

The only workaround that works for me is modifying `PKG_CONFIG_PATH` using the following:
```
export PKG_CONFIG_PATH=/usr/local/lib64/pkgconfig/
```

Then if I `rm -rf build` and execute `meson build`, eveything works OK.

**Comment by nirbheek** [comment]
This is working as expected. pkg-config and ldconfig are not related.

**Comment by c-edw** [comment]
Is setting `PKG_CONFIG_PATH` the intended usage then?

**Comment by nirbheek** [comment]
Yes, that is how you're supposed to tell `pkg-config` about custom pkgconfig prefixes. See: https://people.freedesktop.org/~dbn/pkg-config-guide.html#using which has an example for exactly what you are doing.
